src/entities/mellowLpVault.ts

```diff
@@ -0,0 +1,437 @@
+/* eslint-disable no-console */
+/* eslint-disable camelcase */
+/* eslint-disable lines-between-class-members */
+import {
+  Signer,
+  providers,
+  ethers,
+  BigNumberish,
+  BigNumber,
+  ContractReceipt,
+  Contract,
+} from 'ethers';
+import { isUndefined } from 'lodash';
+import { toBn } from 'evm-bn';
+import { getProtocolPrefix, getTokenInfo } from '../services/getTokenInfo';
+import timestampWadToDateTime from '../utils/timestampWadToDateTime';
+import { getGasBuffer, MaxUint256Bn, TresholdApprovalBn } from '../constants';
+
+import { abi as VoltzVaultABI } from '../ABIs/VoltzVault.json';
+import { abi as Erc20RootVaultABI } from '../ABIs/Erc20RootVault.json';
+import { abi as Erc20RootVaultGovernanceABI } from '../ABIs/Erc20RootVaultGovernance.json';
+import { abi as MarginEngineABI } from '../ABIs/MarginEngine.json';
+import { abi as BaseRateOracleABI } from '../ABIs/BaseRateOracle.json';
+import { abi as IERC20MinimalABI } from '../ABIs/IERC20Minimal.json';
+
+export type MellowLpVaultArgs = {
+  voltzVaultAddress: string;
+  erc20RootVaultAddress: string;
+  erc20RootVaultGovernanceAddress: string;
+  provider?: providers.Provider;
+};
+
+class MellowLpVault {
+  public readonly voltzVaultAddress: string;
+  public readonly erc20RootVaultAddress: string;
+  public readonly erc20RootVaultGovernanceAddress: string;
+  public readonly provider?: providers.Provider;
+
+  public readOnlyContracts?: {
+    marginEngine: Contract;
+    token: Contract;
+    rateOracle: Contract;
+    voltzVault: Contract;
+    erc20RootVault: Contract;
+    erc20RootVaultGovernance: Contract;
+  };
+
+  public writeContracts?: {
+    token: Contract;
+    voltzVault: Contract;
+    erc20RootVault: Contract;
+  };
+
+  public signer?: Signer;
+  public maturity?: string;
+  public protocolId?: number;
+
+  public vaultAccumulative?: number;
+  public vaultCap?: number;
+  public vaultExpectedApy?: number;
+
+  public userDeposit?: number;
```

Comment:
I find the name a bit confusing -- I expext userDeposit to denote the amount of funds deposited by the user. Here rather, userDeposit means the current cashflow of the user, right? If yes, shall we rename it?